xapian-letor/tests/api_letor.cc

```diff
@@ -107,6 +107,8 @@ DEFINE_TESTCASE(preparetrainingfile, generated)
     string qrel = data_directory + "qrel.txt";
     Xapian::prepare_training_file(db_path, query, qrel, 10,
 				  "training_output.txt");
+
+    return true;
```

Comment:
::prepare_training_file() just creates a feature vector corresponding to the <document, query> and appends the document relevance from the qrel file. Checking that it produces the desire output will be basically checking whether the features returned are correct and the correct relevance label is appended. I will be copy pasting the loops thats are implemented in the features to test whether the correct features are returned.